usecases/objects/auto_schema.go

```diff
@@ -492,3 +493,81 @@ func (m *autoSchemaManager) determineNestedPropertiesOfArray(valArray []interfac
 
 	return nestedProperties, nil
 }
+
+type classTenants map[*models.Class]map[string]struct{}
+
+func (m *autoSchemaManager) autoTenants(ctx context.Context,
+	principal *models.Principal, objects []*models.Object,
+) error {
+	var (
+		ct         = make(classTenants)
+		autoTenant = false
+	)
+
+	for _, obj := range objects {
+		// TODO: track the classes we have already checked
+		//       before calling getClass again
+		o := models.Object{Class: obj.Class}
+		class, err := m.getClass(principal, &o)
+		if err != nil {
+			return err
+		}
+
+		// class not found in schema. this will have already
+		// been caught by b.validateObjectsConcurrently, and
+		// will be handled accordingly, so we don't need to
+		// do anything here
+		if class == nil {
+			continue
+		}
+
+		if autoTenant || shouldAutoCreateTenants(class) {
```

Comment:
shouldn't `autoTenant` be set per class? Now it looks like, once `autoTenant` is set it will be set forever, so objects of mixed classes (some with MT enabled, some with MT disabled) will extend `ct` map (though with empty `obj.Tenant` value)

Reply:
Fixed